home/views/api/leaderboard.py

```diff
@@ -0,0 +1,365 @@
+
+#     # Add all accounts found in filtered daily walk data
+#     for email, dw_row in daily_walks.items():
+#         acct = Account.objects.values(*ACCOUNT_FIELDS).get(email=email)
+
+#         # Skip testers unless include_testers
+#         if not include_testers and acct.get("is_tester"):
+#             continue
+
+#         # Don't include those who signed up after the contest ended
+#         if contest and acct["created"] > localize(contest.end):
+#             continue
+
+#         user_stats = user_stats_container.get(
+#             email, dict(new_signup=False)
+#         )
+#         user_stats["account"] = acct
+#         user_stats["dw_steps"] = dw_row["dw_steps"]
+
+#         # Also add recorded walk data
+#         iw_row = intentional_walks.get(email)
+#         if iw_row:
+#             user_stats["rw_steps"] = iw_row["rw_steps"]
+#             user_stats["rw_distance"] = iw_row["rw_distance"]
+#             user_stats["rw_time"] = (
+#                 iw_row["rw_total_walk_time"].total_seconds()
+#                 - iw_row["rw_pause_time"]
+#             ) / 60  # minutes
+#         user_stats["num_rws"] = iw_row["rw_count"] if iw_row else 0
+
+#         # Put user_stats (row) back into container
+#         user_stats_container[email] = user_stats
+
+#     context["user_stats_list"] = user_stats_container.values()
+#     context["contests"] = Contest.objects.all()
+
+#     return context
+
+
+#
+@method_decorator(csrf_exempt, name="dispatch")
+class LeaderboardListView(View):
+    """View to retrieve leaderboard"""
+
+    # model = DailyWalk
+    http_method_names = ["post"]
+
+    def post(self, request, *args, **kwargs):
+        json_data = json.loads(request.body)
+
+        # Validate json. If any field is missing, send back the response message
+        json_status = validate_request_json(
+            json_data,
+            required_fields=["account_id"],
+        )
+        if "status" in json_status and json_status["status"] == "error":
+            return JsonResponse(json_status)
+
+        # Get the device if already registered
+        try:
+            device = Device.objects.get(device_id=json_data["account_id"])
+        except ObjectDoesNotExist:
+            return JsonResponse(
+                {
+                    "status": "error",
+                    "message": (
+                        "Unregistered device - "
+                        f"{json_data['account_id']}."
+                        " Please register first!"
+                    ),
+                }
+            )
+        # Json response template
+        json_response = {
+            "status": "success",
+            "message": "Leaderboard accessed successfully",
+            "payload": {
+                "account_id": device.device_id,
+                "leaderboard": [],
+            },
+        }
+
+        # def get(self, request, *args, **kwargs):
+        # user_stats_container = {}
+
+        contest = Contest.active()
```

Comment:
We can make this API method more generic by passing a contest_id param and using it to fetch the specified Contest. That way, we could have admin dashboards that could show the leaderboard for any contest, active or past...